extensions/scarb-execute/src/oracle/hint_processor.rs

```diff
@@ -0,0 +1,161 @@
+use cairo_lang_casm::hints::{Hint, StarknetHint};
+use cairo_lang_runner::casm_run::{
+    MemBuffer, cell_ref_to_relocatable, extract_relocatable, vm_get_range,
+};
+use cairo_lang_runner::{CairoHintProcessor, insert_value_to_cellref};
+use cairo_vm::Felt252;
+use cairo_vm::hint_processor::hint_processor_definition::{HintProcessorLogic, HintReference};
+use cairo_vm::serde::deserialize_program::ApTracking;
+use cairo_vm::types::exec_scope::ExecutionScopes;
+use cairo_vm::vm::errors::hint_errors::HintError;
+use cairo_vm::vm::errors::vm_errors::VirtualMachineError;
+use cairo_vm::vm::runners::cairo_runner::{ResourceTracker, RunResources};
+use cairo_vm::vm::vm_core::VirtualMachine;
+use std::any::Any;
+use std::collections::HashMap;
+use std::ops::ControlFlow;
+
+pub struct OracleHintProcessor<'a> {
+    pub cairo_hint_processor: CairoHintProcessor<'a>,
+    /// Whether `--experimental-oracles` flag has been enabled.
+    experiment_enabled: bool,
+}
+
+impl<'a> OracleHintProcessor<'a> {
+    /// Creates a new instance of [`OracleHintProcessor`].
+    pub fn new(cairo_hint_processor: CairoHintProcessor<'a>, experiment_enabled: bool) -> Self {
+        Self {
+            cairo_hint_processor,
+            experiment_enabled,
+        }
+    }
+
+    /// Gracefully look if this is one of the cheat codes supported by us.
+    /// Take over execution just right when we learn this is "our" cheat code.
+    /// Any errors will be created by the following [`CairoHintProcessor::execute_hint`] call.
+    fn execute_cheatcode(
+        &mut self,
+        vm: &mut VirtualMachine,
+        _exec_scopes: &mut ExecutionScopes,
+        hint_data: &Box<dyn Any>,
+        _constants: &HashMap<String, Felt252>,
+    ) -> ControlFlow<Result<(), HintError>> {
+        macro_rules! t {
+            ($expr:expr) => {
+                match $expr {
+                    Ok(v) => v,
+                    Err(e) => return ControlFlow::Break(Err(e.into())),
+                }
+            };
+        }
```

Comment:
NIT If it's all `Result<(), HintError>` down the road, maybe it would be more readable if wrapped in a method that returns `Result<(), HintError>` and catched once? 🤔 I feel like it's kind of as if we had a macro that replaces errors with sys::exit.

Reply:
12f5e2d1f6cff1c107c046e0490ed47907197cde